web-ui/src/main/resources/catalog/components/search/map/MapFieldDirective.js

```diff
@@ -134,20 +134,253 @@
+                  // Line2(A,B) and Line1(B) are colinear and Line1(B) lies on Line2
+                  if ((o4 === 0) && onSegment(line2A, line1B, line2B)) return true;
+                  // Nothing intersects
+                  return false;                  
+                }
+
+                /**
+                 * Calculates the intersection point between 'line1' and 'line2'.
+                 * Assumes that the lines actually intersect (use linesIntersect() to verify).
+                 */
+                var getIntersection = function(line1A, line1B, line2A, line2B) {
+                  // Line 1 represented as a1x + b1y = c1 
+                  let a1 = line1B[1] - line1A[1]; 
+                  let b1 = line1A[0] - line1B[0]; 
+                  let c1 = a1 * line1A[0] + b1 * line1A[1]; 
+       
+                  // Line 2 represented as a2x + b2y = c2 
+                  let a2 = line2B[1] - line2A[1]; 
+                  let b2 = line2A[0] - line2B[0]; 
+                  let c2 = a2 * line2A[0] + b2 * line2A[1]; 
+                
+                  let determinant = a1 * b2 - a2 * b1;                
+                  let x = (b2 * c1 - b1 * c2) / determinant; 
+                  let y = (a1 * c2 - a2 * c1) / determinant; 
+                  return [x, y];
+                }
+
+                /**
+                 * Makes a "cut" at the datum line from the geometry edge up or down to the polar coordinate.
+                 * The feature geometry is also transformed to WGS 1984 (EPSG:4326).
+                 * NOTE: This is not suitable for complex or multipart polygons!
+                 * 
+                 *           North pole polygon                   South pole polygon
+                 *          +======+===========+                 +------------------+
+                 *          |      |           |                 |                  |
+                 *          |      o           |                 |            o     |
+                 *          |                  |                 |            |     |
+                 *          +------------------+                 +============+=====+
+                 * 
+                 *   === -> "cut side" 
+                 *   o   -> polar coordinate
+                 * @param {ol.feature} feat
+                 * @param polarCoord
+                 */
+                var modifyPolarExtent = function(feat, polarCoord) {
+                  const atNorthPole = polarCoord == northPole;
+                  const polarLat = atNorthPole ? 90 : -90;
+                  const parts = feat.getGeometry().getCoordinates();
+
+                  // Set the point of origin at equator (where datum line towards polar coordinate starts)
+                  let originCoord = getXY(180, 0);
+
+                  // Show some warnings/info (when in debug mode)
+                  console.log('Search extent contains ' + (atNorthPole ? 'north' : 'south') + ' pole');
```

Comment:
This should be removed

Reply:
Removed? You mean replaced by library calls or moved to a service?

Reply:
No, I mean the `console.log` call should be removed altogether. It can be useful in debug mode but as it is currently it will always show even in production mode.

Reply:
Ah right. Earlier it showed an entire block of code :)
Think I forgot to replace it with `console.debug`. Would that be okay for you or do you really want to remove it altogether?